org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/editor/WNumberEditor.java

```diff
@@ -63,7 +63,16 @@ public class WNumberEditor extends WEditor implements ContextMenuListener
 
     public WNumberEditor()
     {
-    	this("Number", false, false, true, DisplayType.Number, "");
+    	this(DisplayType.Number);
+    }
+
+    /**
+    *
+    * @param displayType
+    */
+    public WNumberEditor(int displayType)
```

Comment:
this would allow new WNumberEditor(DisplayType.Date) which is obvious not correct.

Reply:
nvm that, we already can do WNumberEditor("Date", false, false, true, DisplayType.Date, ""); 
perhaps, possible improvement here is to add displaytype validation to `WNumberEditor(String columnName, boolean mandatory, boolean readonly, boolean updateable, int displayType, String title)`, i.e if (!DisplayType.isNumeric(displayType)) throw new IllegalArgumentException

Reply:
Hi,
Good idea, done